File: TaskMaster_API/TaskMaster/Model/Domain/ProjectData/Taskitem/TaskItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TaskMaster.Model.Domain.ProjectData.Taskitem;

public class TaskItem
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int TaskItemId { get; set; }

    [Required]
    public string Title { get; set; }

    [Required]
    public int Priority { get; set; }
    
    [Required]
    public bool Completed { get; set; }

    [Required]
    public string Notes { get; set; }

    [Required]
    public string Details { get; set; }

    [Required]
    public DateTime DueDate { get; set; }

    public DateTime CompletedDate { get; set; }

    [Required]
    [ForeignKey("Project")]
    public int ProjectId { get; set; }

    [Required
        [ForeignKey("User")]
        public int UserId { get; set; }

    public User User { get; set; }
    public Project Project { get; set; }
}

File: App/TaskMaster/TaskMaster/Models/TaskItem.swift
//
//  TaskItem.swift
//  TaskMaster
//
//  Created by Joshua Root on 12/4/23.
//

import SwiftData
import Foundation

@Model
class TaskItem {
    var title: String
    var priority: Int
    var completed: Bool
    var notes: String
    var details: String
    var dueDate: Date
    var completedDate: Date?
    var project: Project
    
    
    
    init(title: String, priority: Int, completed: Bool, notes: String, details: String, dueDate: Date, completedDate: Date?, project: Project) {
        self.title = title
        self.priority = priority
        self.completed = completed
        self.notes = notes
        self.details = details
        self.dueDate = dueDate
        self.completedDate = completedDate
        self.project = project
    }
    
    
}
